src/main/java/org/reflections/Reflections.java

```diff
@@ -180,8 +180,8 @@ protected void scan() {
             return;
         }
 
-        if (log != null && log.isDebugEnabled()) {
-            log.debug("going to scan these urls:\n" + Joiner.on("\n").join(configuration.getUrls()));
+        if (log != null) {
+            log.debug("going to scan these urls:\n{}", Joiner.on("\n").join(configuration.getUrls()));
```

Comment:
Just noticed that this one is actually a performance degradation:
The Joiner code does get executed (it's not a lambda here) when debug is disabled, while previously it did not. It's only the string concatenation that doesn't happen when debug is disabled due to the {} usage.

Reply:
Ah, true, sorry for that. Should I file a PR to revert this hunk?

Reply:
I think only the removal of the "&& log.isDebugEnabled()" etc from the if's should be enough.
The use of lamdba isn't an option yet due to the below java 8 compatibility needs.